src/load_fund_data.py

```diff
@@ -0,0 +1,233 @@
+    """
+    config = Utils.load_config()
+    database_path = config.database_path
+    input_data_path = config.input_data_path
+    fund_holdings_table_name = TableNames.FUND_HOLDINGS
+    return (database_path, input_data_path, fund_holdings_table_name)
+
+
+def clean_fund_name(fund_name_raw: str) -> str:
+    """
+    Cleans the fund name by removing unwanted characters and converting it to Title case.
+    Assumes that the fund name is the last part of the string after splitting by spaces or special characters.
+    e.g. "TT_monthly_Trustmind", "rpt-Catalysm", "Fund Whitestone", "Applebead"
+
+    Args:
+        fund_name_raw (str): The raw fund name.
+
+    Returns:
+        str: The cleaned fund name.
+    """
+    split_name = re.split(r"[\s\-_.]+", fund_name_raw)
+    end_part = split_name[-1].strip() if split_name else None
+    return end_part.title() if end_part.isalnum() else None
+
+
+def clean_date(date_raw: str) -> str:
+    """
+    Cleans the date string by removing unwanted characters and converting it to a standard format.
+    Supported formats:
+        - "DD-MM-YYYY"
+        - "MM-DD-YYYY"
+        - "DD_MM_YYYY"
+        - "MM_DD_YYYY"
+        - "YYYY-MM-DD"
+        - "YYYYMMDD"
+    Args:
+        date_raw (str): The raw date string.
+
+    Returns:
+        str: The cleaned date string in "YYYY-MM-DD" format.
+    """
+    date_patterns = [
+        (r"\b\d{2}-\d{2}-\d{4}\b", "%d-%m-%Y"),
+        (r"\b\d{2}-\d{2}-\d{4}\b", "%m-%d-%Y"),
+        (r"\b\d{2}_\d{2}_\d{4}\b", "%d_%m_%Y"),
+        (r"\b\d{2}_\d{2}_\d{4}\b", "%m_%d_%Y"),
+        (r"\b\d{4}-\d{2}-\d{2}\b", "%Y-%m-%d"),
+        (r"\b\d{8}\b", "%Y%m%d"),
+    ]
+    for pattern, date_format in date_patterns:
+        date_match = re.search(pattern, date_raw)
+        if date_match:
+            raw_date = date_match.group()
+            try:
+                parsed_date = datetime.strptime(raw_date, date_format)
+                normalized_date = parsed_date.strftime("%Y-%m-%d")
+                return normalized_date
+            except ValueError:
+                continue
+    return None
+
+
+def parse_filename(filename: str) -> tuple[str, str]:
+    """
+    Parses the filename to extract the fund name and date.
+    This assumes the filename is in the format "fund_name_part.date_part".
+
+    Args:
+        filename (str): The filename to parse.
+
+    Returns:
+        tuple[str, str]: A tuple containing the fund name and date.
+
+    Raises:
+        ValueError: If the filename does not contain a valid fund name or date.
+    """
+    base_file_name = os.path.splitext(filename)[0]
+    assert len(base_file_name.split(".")) == 2, f"Invalid filename format: {filename}"
+    fund_name_raw = base_file_name.split(".")[0]
+    date_raw = base_file_name.split(".")[1]
+    fund_name = clean_fund_name(fund_name_raw)
+    date = clean_date(date_raw)
+    if not fund_name or not date:
```

Comment:
This is where you effectively check for None - but you could have allowed a lower-level exception to bubble up.